Source/DafnyCore/AST/Expressions/Datatypes/DatatypeUpdateExpr.cs

```diff
@@ -26,7 +26,11 @@ public class DatatypeUpdateExpr : ConcreteSyntaxExpression, IHasUsages, ICloneab
       Members = original.Members;
       LegalSourceConstructors = original.LegalSourceConstructors;
       InCompiledContext = original.InCompiledContext;
-      ResolvedCompiledExpression = cloner.CloneExpr(original.ResolvedCompiledExpression);
+      if (original.ResolvedExpression == original.ResolvedCompiledExpression) {
```

Comment:
It's so that `ResolvedExpression` and `ResolvedCompiledExpression` can continue to be pointer equal, if they started off that way. And that property is, in turn, used in L58, which decides whether to include just the former or both in the `SubExpressions` (so that `ResolvedCompiledExpression` is not visited twice if it's identical).